sdk/identity/identity/src/util/scopeUtils.ts

```diff
@@ -17,7 +18,7 @@ export function ensureScopes(scopes: string | string[]): string[] {
  */
 export function ensureValidScope(scope: string, logger: CredentialLogger): void {
   if (!scope.match(/^[0-9a-zA-Z-.:/]+$/)) {
-    const error = new Error("Invalid scope was specified by the user or calling client");
+    const error = new CredentialUnavailableError("Invalid scope was specified by the user or calling client");
```

Comment:
I'm not opposed to this change since it seems straightforward enough, but it wouldn't really make sense to me if someone called a valid credential with an invalid scope. 

Is there not a way that we can catch/wrap this inside of a chained credential specifically?

Reply:
Did that. I think that works too!